lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,436 @@
+use bytes::Bytes;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use lookup::path;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use smallvec::{smallvec, SmallVec};
+use std::collections::HashMap;
+use value::Kind;
+use vector_core::{
+    config::{log_schema, DataType},
+    event::Event,
+    schema,
+};
+
+use super::Deserializer;
+
+/// GELF Message fields. Definitions from https://docs.graylog.org/docs/gelf
+pub mod gelf_fields {
```

Comment:
I'm not seeing these fields used outside of this file. Does it need to be `pub`?

Reply:
Ah, in earlier commits the unit tests were in a separate file.
However, in the upcoming GELF encoder PR, I will be referencing these fields in `lib/codecs/src/encoding/format/gelf.rs`

Reply:
I considered that it might be better to put the `gelf_fields` module above, like somewhere in `lib/codecs/src`

Reply:
Mmm, yeah, I'd be 👍 on moving these shared types up into `lib/codecs/gelf.rs` (keeping the decoder and encoder in their respective modules).